src/core/model/Stroke.cpp

```diff
@@ -72,6 +78,68 @@ auto Stroke::cloneStroke() const -> Stroke* {
 
 auto Stroke::clone() -> Element* { return this->cloneStroke(); }
 
+std::unique_ptr<Stroke> Stroke::cloneSection(const PathParameter& lowerBound, const PathParameter& upperBound) const {
+    auto s = std::make_unique<Stroke>();
+    s->applyStyleFrom(this);
+    s->width = this->width;  // stroke width, not bounding box width
+
+    if (upperBound.index == lowerBound.index) {
+        const Point& p = this->getPoint(lowerBound);
+        const Point& q = this->getPoint(upperBound);
+        s->points = {p, q};
+    } else {
```

Comment:
Assuming the function is used correct, can it happen, that this case isn't handled also by the else branch?
In my opinion, `upperBound.t != 0.0` should never be true.

Reply:
I assume you meant `upperBound.t == 0.0`, right? Even then, there is (at least) one case where it can happen: `lowerBound = (0, 0.0)` and `upperBound = (0, 0.0)`. This is obviously a corner case, but still.

Also, I thought having this function (and the other similar functions like `ErasableStrokeView::draw`) as generic as possible would limit the number of corner-case-tests when computing those bounds.

Reply:
You're right this case could be dealt with by the `else` branch (thus removing the if statement), removing the `if (upperBound.t != 0.0)` below, and asserting that `upperBound.t` can only be 0.0 if `lowerBound == upperBound`.